src/control/server/engine/config.go

```diff
@@ -422,6 +424,38 @@ func (c *Config) UpdatePMDKEnvars() error {
 	return nil
 }
 
+// Increase ABT stack size for UCX provider.
+func (c *Config) UpdateABTEnvarsUCX() error {
+
+	providerStr, err := c.Fabric.GetPrimaryProvider()
+	if err != nil {
+		return err
+	}
+
+	if strings.Index(providerStr, "ucx") == -1 {
+		return nil
+	}
+
+	stackSizeStr, err := c.GetEnvVar("ABT_THREAD_STACKSIZE")
+	if err != nil {
```

Comment:
Would be nice to check for some type of "NOTEXIST" error here if that can be done Rather than just assuming that any error indicates the absence of the env.

Reply:
I checked the implementation of GetEnvVar function, and it can only return non-nil if the environment variable does not exist - no other reason for an error.